### PR TITLE
Fix image segmentation pipeline errors, resolve backward compatibility issues

### PR DESCRIPTION
**Pull request by alaradirik**
# What does this PR do?
- Replaces hard-coded mask score binarization threshold used in `post_process_panoptic_segmentation` and `post_process_instance_segmentation` method of DETR and MaskFormer feature extractors with the `mask_threshold` argument
- Adds copied from statements to MaskFormer post processing helper functions
- Reintroduces the mask_threshold argument to the `ImageSegmentationPipeline` (resolves backward compatibility issues)
- Renames `ImageSegmentationPipeline` task argument to subtask to avoid conflicts, defaults to None
- If `ImageSegmentationPipeline` subtask is unset, tries performing panoptic segmentation first, followed by instance and semantic segmentation
- Fixes broken `ImageSegmentationPipeline` tests

Reintroducing the `mask_threshold` argument and setting it to 0 fixes the test errors and ensures at least one mask is returned for any input and model.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [X ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [X ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: tests/pipelines/test_pipelines_image_segmentation.py [hunk 1]
```diff
@@ -147,60 +154,39 @@ def test_small_model_tf(self):
         pass
 
     @require_torch
+    @unittest.skip("No weights found for hf-internal-testing/tiny-detr-mobilenetsv3-panoptic")
```
Comment (Narsil) [inline tests/pipelines/test_pipelines_image_segmentation.py:157]: No this test should be enabled.

**Review by sgugger — APPROVED** [review]
Thanks for the fixes!

File: src/transformers/pipelines/image_segmentation.py [hunk 2]
```diff
@@ -116,11 +120,17 @@ def _forward(self, model_inputs):
         model_outputs["target_size"] = target_size
         return model_outputs
 
-    def postprocess(self, model_outputs, task="semantic", threshold=0.9, overlap_mask_area_threshold=0.5):
-        if task == "instance" and hasattr(self.feature_extractor, "post_process_instance_segmentation"):
+    def postprocess(
+        self, model_outputs, subtask=None, threshold=0.9, mask_threshold=0.5, overlap_mask_area_threshold=0.5
+    ):
+        print(threshold, mask_threshold, overlap_mask_area_threshold)
```
Comment (amyeroberts) [inline src/transformers/pipelines/image_segmentation.py:126]:
```suggestion
```

**Review by amyeroberts — APPROVED** [review]
Thanks for this fix! LGTM :)

**Pull request merged**